src/atsamd/fdcan.c

```diff
@@ -234,13 +249,18 @@ compute_btr(uint32_t pclock, uint32_t bitrate)
 void
 can_init(void)
 {
+#if CONFIG_HAVE_SAMD_USB
     if (!CONFIG_USB) {
         // The FDCAN peripheral only seems to run if at least one
         // other peripheral is also enabled.
         enable_pclock(USB_GCLK_ID, ID_USB);
         USB->DEVICE.CTRLA.reg = USB_CTRLA_ENABLE;
     }
+#endif
 
+#if CONFIG_MACH_SAMC21
+    MCLK->AHBMASK.reg |= MCLK_AHBMASK_CANx;
+#endif
```

Comment:
Can you explain what this does?

Reply:
This bus clock for can is not enabled by default on this chip (annoyingly, see section 17.8.6 of the data sheet). This turns it on. Notice that it's not the same as the one in the clock code.